src/go/rpk/pkg/cli/transform/list_test.go

```diff
@@ -0,0 +1,135 @@
+package transform
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+	"unicode"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"gopkg.in/yaml.v3"
+
+	"github.com/stretchr/testify/require"
+)
+
+func setupTestData() []adminapi.TransformMetadata {
+	return []adminapi.TransformMetadata{
+		{
+			Name:         "foo2bar",
+			InputTopic:   "foo",
+			OutputTopics: []string{"bar"},
+			Environment: []adminapi.EnvironmentVariable{
+				{Key: "FLUBBER", Value: "enabled"},
+			},
+			Status: []adminapi.PartitionTransformStatus{
+				{
+					NodeID:    0,
+					Partition: 1,
+					Core:      0,
+					Status:    "running",
+				},
+				{
+					NodeID:    0,
+					Partition: 2,
+					Core:      1,
+					Status:    "running",
+				},
+				{
+					NodeID:    1,
+					Partition: 3,
+					Core:      4,
+					Status:    "inactive",
+				},
+			},
+		},
+		{
+			Name:         "scrubber",
+			InputTopic:   "pii",
+			OutputTopics: []string{"cleaned", "munged"},
+			Environment: []adminapi.EnvironmentVariable{
+				{Key: "FLUX_CAPACITOR", Value: "disabled"},
+			},
+			Status: []adminapi.PartitionTransformStatus{
+				{
+					NodeID:    0,
+					Partition: 1,
+					Core:      0,
+					Status:    "errored",
+				},
+			},
+		},
+	}
+}
+
+func TestPrintSummaryView_Text(t *testing.T) {
+	s := summarizedView(setupTestData())
+	f := config.OutFormatter{Kind: "text"}
+	b := &strings.Builder{}
+	printSummary(f, s, b)
+	expected := `
+NAME      INPUT TOPIC  OUTPUT TOPIC     RUNNING
+foo2bar   foo          bar              2 / 3
+scrubber  pii          cleaned, munged  0 / 1
+`
+	require.Equal(t, strings.TrimLeftFunc(expected, unicode.IsSpace), b.String())
+}
+
+func TestPrintSummaryView_Json(t *testing.T) {
+	s := summarizedView(setupTestData())
+	f := config.OutFormatter{Kind: "json"}
+	b := &strings.Builder{}
+	printSummary(f, s, b)
+	expected, err := json.Marshal(s)
+	require.NoError(t, err)
+	require.Equal(t, string(expected)+"\n", b.String())
+}
+
+func TestPrintSummaryView_Yaml(t *testing.T) {
+	s := summarizedView(setupTestData())
+	f := config.OutFormatter{Kind: "yaml"}
+	b := &strings.Builder{}
+	printSummary(f, s, b)
+	expected, err := yaml.Marshal(s)
+	require.NoError(t, err)
+	require.Equal(t, string(expected)+"\n", b.String())
+}
```

Comment:
[nit] We usually prefer table-driven tests. But I guess it's fine given that we only have 2 kinds.

Reply:
Happy to refactor would expect:

- A single test with parameters of `kind`, `summary/detail`
- Two tests with parameter of `kind`
- 3 tests with parameter of `summary/detail`

Reply:
Either 1 or 2 it's fine for me :+1:

Reply:
I added a test for summary and a test for detail then used table driven tests for each format output.